returnn/tf/layers/rec.py

```diff
@@ -4874,7 +4874,9 @@ def update_var():
         s.set_shape(shape_invariant)
         return s
     elif initial_state == "placeholder":
-      ph = tf_compat.v1.placeholder(tf.float32, shape=shape_invariant, name="placeholder_%s" % key_name)
+      assert rec_layer is not None
+      with rec_layer.var_creation_scope():
+        ph = tf_compat.v1.placeholder(tf.float32, shape=shape_invariant, name="placeholder_%s" % key_name)
```

Comment:
I wonder if we should use a better name, to make it less ambiguous. Sth like `state_placeholder_...`.
```suggestion
        ph = tf_compat.v1.placeholder(
          tf.float32, shape=shape_invariant, name="state_placeholder_%s" % key_name)
```